kernel/src/table_changes/mod.rs

```diff
@@ -0,0 +1,196 @@
+//! Provides an API to read the table's change data feed between two versions.
+use std::sync::LazyLock;
+
+use url::Url;
+
+use crate::log_segment::LogSegment;
+use crate::path::AsUrl;
+use crate::schema::{DataType, Schema, StructField, StructType};
+use crate::snapshot::Snapshot;
+use crate::table_features::ColumnMappingMode;
+use crate::{DeltaResult, Engine, Error, Version};
+
+static CDF_FIELDS: LazyLock<[StructField; 3]> = LazyLock::new(|| {
+    [
+        StructField::new("_change_type", DataType::STRING, false),
+        StructField::new("_commit_version", DataType::LONG, false),
+        StructField::new("_commit_timestamp", DataType::TIMESTAMP, false),
+    ]
+});
+
+/// Represents a call to read the Change Data Feed between two versions of a table. The schema of
+/// `TableChanges` will be the schema of the table with three additional columns:
+/// - `_change_type`: String representing the type of change that for that commit. This may be one
+///   of `delete`, `insert`, `update_preimage`, or `update_postimage`.
+/// - `_commit_version`: Long representing the commit the change occurred in.
+/// - `_commit_timestamp`: Time at which the commit occurred. If In-commit timestamps is enabled,
+///   this is retrieved from the [`CommitInfo`] action. Otherwise, the timestamp is the same as the
+///   commit file's modification timestamp.
+///
+/// For more details, see the following sections of the protocol:
+/// - [Add CDC File](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#add-cdc-file)
+/// - [Change Data Files](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#change-data-files).
+///
+/// [`CommitInfo`]: crate::actions::CommitInfo
+#[derive(Debug)]
+pub struct TableChanges {
+    pub log_segment: LogSegment,
+    table_root: Url,
+    end_snapshot: Snapshot,
+    start_version: Version,
```

Comment:
Just checking, we won't have to re-find the start snapshot when we actually go to return results right?

Reply:
No, we don't. Henceforth all we are interested in are the commits in the `LogSegment`, and the schema we got from the end_snapshot.